account_bank_statement_import_adyen/models/account_bank_statement_import.py

```diff
@@ -0,0 +1,135 @@
+# © 2017 Opener BV (<https://opener.amsterdam>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from io import BytesIO
+from openpyxl import load_workbook
+from zipfile import BadZipfile
+
+from odoo import api, models, fields
+from odoo.exceptions import UserError
+from odoo.tools.translate import _
+
+
+class AccountBankStatementImport(models.TransientModel):
+    _inherit = 'account.bank.statement.import'
+
+    @api.model
+    def _parse_file(self, data_file):
+        """Parse an Adyen xlsx file and map merchant account strings
+        to journals. """
+        try:
+            return self.import_adyen_xlsx(data_file)
+        except ValueError:
+            return super(AccountBankStatementImport, self)._parse_file(
+                data_file)
+
+    def _find_additional_data(self, currency_code, account_number):
+        """ Try to find journal by Adyen merchant account """
+        if account_number:
+            journal = self.env['account.journal'].search([
+                ('adyen_merchant_account', '=', account_number)], limit=1)
+            if journal:
+                if self._context.get('journal_id', journal.id) != journal.id:
+                    raise UserError(
+                        _('Selected journal Merchant Account does not match '
+                          'the import file Merchant Account '
+                          'column: %s') % account_number)
+                self = self.with_context(journal_id=journal.id)
+        return super(AccountBankStatementImport, self)._find_additional_data(
+            currency_code, account_number)
+
+    @api.model
+    def balance(self, row):
+        return -(row[15] or 0) + sum(
```

Comment:
If row position 15,16, 17, 18, 19, 20 is necessary add float conversion

return -(float(row[15]) if row[15] else 0.0) + sum(
            float(row[i]) if row[i] else 0.0
            for i in (16, 17, 18, 19, 20))